### PR TITLE
fix(auth): add unauthorized-route for users without login information

### PR DESCRIPTION
**Pull request by nikomakela**
KK-1063. KK-1107.
If user is unauthorized to view a route, if he is logged in, he will be taken to the (old) wrong-login-method page, but if he isn't, then he is taken to a new unauthorized-route. After the login, the user will be taken to the URL he was trying to open at the first place.

Testing instructions: https://github.com/City-of-Helsinki/kukkuu-ui/pull/537#issuecomment-1988588085

The (updated) view:
image

### CODE REVIEW

**Comment by nikomakela** [comment]
### How to test
First, get the actual URLs you would like to test:

Login to the Kukkuu UI

Navigate to the page that you want to test

Copy the URL to some notepad

Then, logout and try to access the URL without logging in. You should be taken to a new /unauthorized path and the URL you tried to access should be set as a search parameter (in the URL). Then, use the login-button and login. You should now be taken to the actual page you tried to access at teh first place.

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr537.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr537.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr537.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr537.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by nikomakela** [comment]
The new route view was first looking like this
image

but it was changed to this from a request of product owner
image

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr537.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr537.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Review by karisal-anders — APPROVED** [review]

File: src/domain/app/layout/utilityComponents/InfoTemplate.tsx [hunk 1]
```diff
@@ -1,31 +1,40 @@
...
-    <div className={styles.infoPageLayout}>
+    <div className={classnames(styles.infoPageLayout, classes)}>
       <h1 className={styles.infoPageLayoutTitle}>{title}</h1>
-      <Icon className={styles.infoPageLayoutFace} src={icon} />
-      <p className={styles.infoPageLayoutDescription}>{description}</p>
+      {typeof icon === 'string' ? (
+        <Icon className={styles.infoPageLayoutFace} src={icon} />
+      ) : (
+        icon
+      )}
```
Comment (karisal-anders) [inline src/domain/app/layout/utilityComponents/InfoTemplate.tsx:34]: Out of curiosity, was there something broken before? I don't see any changes in this PR using InfoTemplate with ReactElement, only string. So are you just making this more futureproof so to speak?

Comment (nikomakela) [reply src/domain/app/layout/utilityComponents/InfoTemplate.tsx:34]: Yep, "futureproof". I tested a HDS icon there, but wasn't happy with it.

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr537.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr537.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Pull request merged**